backend/src/main/java/com/stock/admin/controller/ProductController.java

```diff
@@ -46,59 +56,50 @@ public Response getAllProducts(@RequestParam(required = false, name = "shopCode"
      * @param productCode the product code
      * @return the products details
      */
-    @RequestMapping(value = "/{productCode}", method = RequestMethod.GET, produces = "application/json; charset=UTF-8")
+    @GetMapping(value = "/{productCode}")
     @ResponseBody
-    public Response getProductsDetails(@PathVariable("productCode") String productCode) throws Exception {
+    public Response getProductsDetails(@PathVariable("productCode") String productCode) {
         Optional<Product> product = productService.getByProductCode(productCode);
         return product.map(p -> Response.buildResponse(Product.type, p, true))
-                .orElseThrow(() -> new Exception(HttpStatus.NOT_FOUND.getReasonPhrase()));
-
+                .orElseThrow(() -> new StockAdminApplicationException("Product does not exists", HttpStatus.NOT_FOUND));
     }
 
-
     /**
      * Add product product response.
      *
      * @param productRequest the create product request
      * @return the product response
      */
-    @RequestMapping(method = RequestMethod.POST, produces = "application/json; charset=UTF-8")
+    @PostMapping
     @ResponseBody
-    public Response addProduct(@RequestBody ProductRequest productRequest) {
+    public Response addProduct(@Valid @RequestBody ProductRequest productRequest) {
+        productRequest.getProduct().setCode(sequenceGeneratorService.generateSequence(Product.SEQUENCE_NAME));
         Product addedProduct = productService.create(productRequest.getProduct());
-        Response productResponseBody = new Response();
-        productResponseBody.setPayload(addedProduct);
-        productResponseBody.setStatus(ResponseStatus.SUCCESS);
-        return productResponseBody;
+        return Response.buildResponse(Product.type, addedProduct, true);
     }
 
-
     /**
      * Modify by product by id product response.
      *
      * @param productCode    the product code
      * @param productRequest the product request
      * @return the product response
      */
-    @RequestMapping(value = "/{id}", method = RequestMethod.PUT)
+    @PutMapping(value = "/{id}")
     @ResponseBody
     public Response modifyByProductById(@PathVariable("id") String productCode, @Valid @RequestBody ProductRequest productRequest) {
```

Comment:
You can try removing `@valid` from here, so that validation does not trigger for update case.

And can manually do validations required for update.